### PR TITLE
Better handling of guests in multiplayer lobby

### PR DESCRIPTION
**Pull request by AlexNisnevich**
- Guests have better-looking usernames.
- Guests can join and host games and everything works correctly.
- Guests can no longer join the matchmaking queue.
- Your username is italicized in the player list to make it easier to tell which guest you are.
- Fixed some bugs.

### CODE REVIEW

File: src/common/components/play/multiplayer/LobbyStatus.js [hunk 1]
```diff
@@ -22,11 +22,13 @@ const LobbyStatus = (props) => {
   return (
     <Paper style={{padding: 20, marginBottom: 20}}>
       <div style={{position: 'relative'}}>
-        { connected ? <span>
+        { connected ? <span className="players-online">
```
Comment (jacobnisnevich) [inline src/common/components/play/multiplayer/LobbyStatus.js]: Something like:

```
const styles = {
  playersOnline: {
    'span:after': {
      content: ','
    }
  }
};
```

and then

```
<span className={classes.playersOnline} />
```

and

```
export default withStyles(styles)(LobbyStatus);
```

would be more in line with MUI v1.x

File: src/common/reducers/socket.js [hunk 2]
```diff
@@ -17,21 +17,25 @@ export default function socket(oldState = cloneDeep(defaultState), action) {
...
       const message = {
-        user: action.payload.sender ?
-          (state.userDataByClientId[action.payload.sender] ?
-            state.userDataByClientId[action.payload.sender].displayName :
-            action.payload.sender) :
-          'You',
-        text: action.payload.msg,
+        user: sender ?
+                (userDataMap[sender] ? userDataMap[sender].displayName : sender) :
+                'You',
```
Comment (jacobnisnevich) [inline src/common/reducers/socket.js]: Should `'You'` be a constant?

File: src/common/components/play/multiplayer/LobbyStatus.js (hunk 1 above)
Comment (AlexNisnevich) [reply src/common/components/play/multiplayer/LobbyStatus.js]: What about the `.players-online span:last-of-type::after` CSS rule? Could I make that work as well using MUI v1.x styling?

File: src/common/reducers/socket.js (hunk 2 above)
Comment (AlexNisnevich) [reply src/common/reducers/socket.js]: I guess I could make a `YOU_PLAYER_NAME` constant or something but I think it's pretty obvious what its value would be?

File: src/common/components/play/multiplayer/LobbyStatus.js (hunk 1 above)
Comment (jacobnisnevich) [reply src/common/components/play/multiplayer/LobbyStatus.js]: Yup same idea

File: src/common/reducers/socket.js (hunk 2 above)
Comment (jacobnisnevich) [reply src/common/reducers/socket.js]: What if it changes :O

Comment (AlexNisnevich) [reply src/common/reducers/socket.js]: uhhhh

**Pull request merged**